src/utils/src/pid.c

```diff
@@ -73,15 +78,23 @@ float pidUpdate(PidObject* pid, const float measured, const bool updateError)
   * in the setpoint. By using the process variable for the derivative calculation, we achieve
   * smoother and more stable control during setpoint changes.
   */
-  float deriv = -(measured - pid->prevMeasured) / pid->dt;
+  float delta = -(measured - pid->prevMeasured);
+
+  if (isYawAngle){
```

Comment:
Consider adding a comment to explain that this code prevents D spikes.  It’s amusing that the workaround for derivative kicks ended up introducing kicks in the derivative itself.

Reply:
Perhaps include this in the note above

Reply:
I added a comment about the yaw spike fix

Reply:
> It’s amusing that the workaround for derivative kicks ended up introducing kicks in the derivative itself.

Thanks for the review, yes, indeed :D